# **<center style="color: green;">Desafío - Segmentación de Clientes</center>**

#### **Nombre:** Cristian Tobar Morales ###
#### **Curso:** Machine Learning ###

# **Descripción**
En el apasionante mundo del comercio minorista, el supermercado "FreshMart" se ha
convertido en el lugar preferido para realizar las compras del día a día. Con el objetivo de
brindar una experiencia de compra más personalizada, "FreshMart" ha decidido emplear
técnicas de aprendizaje no supervisado para segmentar a sus clientes y diseñar estrategias
de marketing adaptadas a sus necesidades y preferencias.

Con un amplio conjunto de datos recopilados durante el último año, que incluyen
información de compras y perfiles de clientes, "FreshMart" cuenta con una valiosa fuente de
información para el análisis. Ahora, el desafío radica en aplicar técnicas de clustering
innovadoras para descubrir segmentos de clientes que comparten patrones de compra
similares y comportamientos de consumo.

## objetivo principal de la segmentación de clientes
Identificar grupos específicos dentro
de la base de clientes de "FreshMart" que puedan beneficiarse de promociones y campañas
de marketing personalizadas. La idea es poder caracterizar estos segmentos y ponerle un
nombre a cada uno.

## objetivo final de "FreshMart"
Es mejorar la satisfacción del cliente y optimizar la
rentabilidad del supermercado. Al emplear técnicas de clustering avanzadas y aprovechar
los datos históricos disponibles, "FreshMart" estará en condiciones de ofrecer una
experiencia de compra más personalizada y adaptada a las necesidades de cada segmento
de clientes. Esto consolida la fidelidad de los clientes existentes y atrae a nuevos clientes
que buscan un supermercado que se preocupe por sus preferencias individuales.

## Descripción de los atributos 
La data entregada esta a nivel de boleta y tiene los siguientes campos:
- `InvoiceNo:` Nro identificador de cada boleta
- `InvoiceDate:` Fecha de la boleta
- `CustomerId:` Nro identificador de cada usuario
- `Quantity:` Cantidad de unidades totales compradas
- `price_total:` Precio total de la boleta
- `StockCode:` Cantidad de unidades diferentes compradas

# **Desarrollo**

## Cargar el DataSet y Exploración inicial

### Importar librerías 

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

### Cargar el dataset y exploración

In [3]:
# Cargar el archivo
df = pd.read_excel('Retail_Invoices.xlsx', engine='openpyxl')

# Mostrar el DataSet
df.head()

InvoiceNo InvoiceDate  CustomerID  Quantity  price_total  StockCode
0    536365  2010-12-01       17850        40       139.12          7
1    536366  2010-12-01       17850        12        22.20          2
2    536367  2010-12-01       13047        83       278.73         12
3    536368  2010-12-01       13047        15        70.05          4
4    536369  2010-12-01       13047         3        17.85          1

In [4]:
# Información del dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22190 entries, 0 to 22189
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   InvoiceNo    22190 non-null  object        
 1   InvoiceDate  22190 non-null  datetime64[ns]
 2   CustomerID   22190 non-null  int64         
 3   Quantity     22190 non-null  int64         
 4   price_total  22190 non-null  float64       
 5   StockCode    22190 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(3), object(1)
memory usage: 1.0+ MB


Con respecto a la información inicial del dataset se puede concluir:
- La cantidad de columnas: 6
- Cantidad de filas: 22190
- EL tipo de datos corresponde: datetime64(1), float64(1), int64(3), object(1)
- La data no cuenta con valores nulos